Reference/Management/Services/MediaService/Index.md

```diff
@@ -106,3 +127,29 @@ When creating a new media from a file (eg. of the types **Image** or **File**),
 
 Umbraco uses this instance to determine the type of the media you're creating, as well as handling a few things "under the hood" so you don't have to. For instance Umbraco will automatically set other properties relatd to the file - such as file size and image dimensions.
 :::
+
+
+### Creating a new media from a stream
+
+Similar to specifying a `HttpPostedFileBase`as shown in the example above, you can instead specify a `Stream` for the contents of the file that should be created.
+
+As an example, if you have a file on disk, you can open a new stream for a file on the disk, and then create a new media for that file in Umbraco:
+
+```csharp
+// Open a new stream to the file
+using (Stream stream = File.OpenRead("C:/path/to/my-image.jpg"))
+{
+
+    // Initialize a new image at the root of the media archive
+    IMedia media = Services.MediaService.CreateMedia("My image", Constants.System.Root, Constants.Conventions.MediaTypes.Image);
+
+    // Set the property value (Umbraco will handle the underlying magic)
```

Comment:
Is it really magic though? 😅 
```suggestion
    // Set the property value (Umbraco will handle the underlying magic)
```

Reply:
I've now changed the text to "Umbraco will automatically update related properties" 😉